openghg/analyse/_scenario.py

```diff
@@ -1287,102 +1285,78 @@ def _calc_modelled_obs_HiTRes(
             # Reindex to match to correct values
             flux_ds_high_freq = flux_ds_high_freq.reindex({"time": full_dates}, method="ffill")
         elif flux_res_H > 24:
+            # TODO this case should be handled outside of the "compute_fp_x_flux" function
             # If flux is not high frequency use the monthly averages instead.
             flux_ds_high_freq = flux_ds_low_freq
 
         # TODO: Add check to make sure time values are exactly aligned based on date range
 
-        # Make sure the dimensions match the expected order for indexing
-        fp_HiTRes = fp_HiTRes.transpose(*("lat", "lon", "time", "H_back"))
-        flux_ds_high_freq = flux_ds_high_freq.transpose(*("lat", "lon", "time"))
+        # Extract flux data from dataset
+        flux_high_freq = flux_ds_high_freq.flux
+        flux_low_freq = flux_ds_low_freq.flux
+
+        def make_hf_flux_rolling_avg_array(
+            flux_high_freq: xr.DataArray,
+            fp_high_time_res: xr.DataArray,
+            highest_res_H: int,
+            max_h_back: int,
+        ) -> xr.DataArray:
+            # create windows (backwards in time) with `max_h_back` many time points,
+            # starting at each time point in flux_hf_rolling.time
+            window_size = max_h_back // highest_res_H
+            flux_hf_rolling = flux_high_freq.rolling(time=window_size).construct("H_back")
+
+            # set H_back coordinates using highest_res_H frequency
+            h_back_type = fp_high_time_res.H_back.dtype
+            flux_hf_rolling = flux_hf_rolling.assign_coords(
+                {"H_back": np.arange(0, max_h_back, highest_res_H, dtype=h_back_type)[::-1]}
+            )
 
-        # Extract footprints array to use in numba loop
-        fp_HiTRes = da.array(fp_HiTRes)
+            # select subsequence of H_back times to match high res fp (i.e. fp without max H_back coord)
+            flux_hf_rolling = flux_hf_rolling.sel(H_back=fp_high_time_res.H_back)
```

Comment:
The re-ordering is probably just incidental. The coordinates need to be assigned in the reverse order on lines 1311-1312 because the rolling window goes backwards in time. The footprint `H_back` coordinate is in increasing order, so that re-orders the `H_back` coordinate of the rolling window.

The main purpose of this part of the code is to select of a subset of the times, like you say in part (b).

I didn't think to mention the reordering in the comments because with xarray you don't need to worry about that.  (But the reverse order on lines 1311-1312 is important.)

I guess this could be confusing if a subset of `H_back` doesn't need to be selected. Also I guess the comment below the signature is slightly inaccurate since the number of times points could change at line 1316.

The selection part could be moved nearer to the calculation on line 1343. (I guess another option could be to fill values in the footprint to match the flux, if the flux is higher frequency. I'm not sure if that is better or worse than dropping the extra flux values in this case.

Reply:
Thanks for clarifying and that all sounds fine - I think initially I just wondered on this because of the clause "(i.e. fp without max H_back coord)" but flux_hf_rolling already didn't include the max H_back. However, re-reading the comment I think I now understand the meaning better with "select subsequence" as the key part of the comment. Before it seemed like there was an expectation that the max H_back coord would be within flux_hf_rolling but I think I understand the "i.e." clause better now.

I don't think any changes are needed (I think it's just complex!).

Reply:
Ah yea, the "i.e." was just a reminder of what "high res fp" is supposed to mean. That should be clearer with the new footprint format.